packages/prompts/test/autocomplete.test.ts

```diff
@@ -0,0 +1,124 @@
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import { autocomplete } from '../src/autocomplete.js';
+import { MockReadable, MockWritable } from './test-utils.js';
+
+describe('autocomplete', () => {
+	let input: MockReadable;
+	let output: MockWritable;
+	const testOptions = [
+		{ value: 'apple', label: 'Apple' },
+		{ value: 'banana', label: 'Banana' },
+		{ value: 'cherry', label: 'Cherry' },
+		{ value: 'grape', label: 'Grape' },
+		{ value: 'orange', label: 'Orange' },
+	];
+
+	beforeEach(() => {
+		input = new MockReadable();
+		output = new MockWritable();
+	});
+
+	afterEach(() => {
+		vi.restoreAllMocks();
+	});
+
+	test('renders initial UI with message and instructions', async () => {
+		const result = autocomplete({
```

Comment:
we should probably make sure these get disposed properly

e.g.

```ts
const result = autocomplete(...);

input.emit('key', '', {name: 'return'});
await result;
```

just so we don't end up with loads of dangling promises all over